gdnative-core/src/core_types/string.rs

```diff
@@ -260,6 +260,36 @@ impl std::hash::Hash for GodotString {
     }
 }
 
+impl Add<&GodotString> for &GodotString {
+    type Output = GodotString;
+    fn add(self, other: &GodotString) -> GodotString {
+        GodotString::from_sys(unsafe { (get_api().godot_string_operator_plus)(&self.0, &other.0) })
+    }
+}
+
+impl AddAssign<&GodotString> for GodotString {
+    fn add_assign(&mut self, other: &Self) {
+        self.0 = unsafe { (get_api().godot_string_operator_plus)(&self.0, &other.0) };
+    }
+}
+
+impl PartialOrd for GodotString {
+    fn partial_cmp(&self, other: &GodotString) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for GodotString {
+    fn cmp(&self, other: &GodotString) -> Ordering {
+        let less = unsafe { (get_api().godot_string_operator_less)(&self.0, &other.0) };
+        match (less, self == other) {
+            (true, false) => Ordering::Less,
+            (false, false) => Ordering::Greater,
+            (_, true) => Ordering::Equal,
+        }
```

Comment:
NIT: This will always do two comparisons. So maybe doing the equality check first and only then doing `<` might be a bit better?

```suggestion
        if self == other {
            Ordering::Equal
        } else if unsafe { (get_api().godot_string_operator_less)(&self.0, &other.0) } {
            Ordering::Less
        } else {
            Ordering::Greater
        }
```

Reply:
Sounds good, I think I'll make it do the less than first, just because chances are that someone using a `cmp/</>` is probably expecting different strings, otherwise they would probably be using `eq/==`.

Reply:
`String::operator<` actually performs a strict less-than comparison, so `==` shouldn't be checked when it returns `true`.